Content.Shared/Bed/Sleep/SleepingSystem.cs

```diff
@@ -328,4 +333,23 @@ public sealed partial class WakeActionEvent : InstantActionEvent;
 /// Raised on an entity when they fall asleep or wake up.
 /// </summary>
 [ByRefEvent]
-public record struct SleepStateChangedEvent(bool FellAsleep);
+public sealed class SleepStateChangedEvent : EntityEventArgs
+{
+    public bool FellAsleep;
+
+    /// <summary>
+    /// The amount of time this entity slept for. Null if <see cref="FellAsleep"/> is true.
+    /// </summary>
+    public TimeSpan? TimeSlept;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="SleepStateChangedEvent"/> class.
+    /// </summary>
+    /// <param name="fellAsleep">True if the entity fell asleep, false if it woke up.</param>
+    /// <param name="timeSlept">The duration of sleep, or null if the entity just fell asleep.</param>
+    public SleepStateChangedEvent(bool fellAsleep, TimeSpan? timeSlept = null) // DeltaV
+    {
+        FellAsleep = fellAsleep;
+        TimeSlept = timeSlept;
+    }
```

Comment:
previous review still stands, keep this a struct and just add a field to the list